src/tn_sdk/tests/test_prepare_data_for_generate_solutions.py

```diff
@@ -0,0 +1,33 @@
+import unittest
+import zlib
+from base64 import b64encode
+
+from src.tn_sdk import TnApi
+from src.tn_sdk.exceptions.exceptions import InvalidDataException
+
+
+class TestPrepareDataForGenerateSolutions(unittest.TestCase):
+    def setUp(self):
+        """Create a reusable client instance for tests."""
+        self.api_prod = TnApi("abc123", "refresh456", is_sandbox=False)
+
+    def test_prepare_data_for_generate_solutions__valid_data__returns_bytes(self):
+        input_data = '{"test": "value"}'
```

Comment:
super minor - to be super explicit could start with a python dictionary and then json.dumps here instead of starting with a json string.